src/nnet3/nnet-simple-component.cc

```diff
@@ -2082,9 +2086,12 @@ void Convolutional1dComponent::InitFromConfig(ConfigLine *cfl) {
   int32 input_dim = -1, output_dim = -1;
   int32 patch_dim = -1, patch_step = -1, patch_stride = -1;
   cfl->GetValue("learning-rate", &learning_rate); //optional
+  cfl->GetValue("patch-dim", &patch_dim);
```

Comment:
Please modify to 
<pre><code>
ok = ok && cfl->GetValue("patch-dim", &patch_dim);
ok = ok && cfl->GetValue("patch-step", &patch_step);
ok = ok && cfl->GetValue("patch-stride", &patch_stride);
</code></pre>

the method has to ensure that the specified values exist.